stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api/RetrofitCallAdapterFactory.kt

```diff
@@ -24,21 +28,31 @@ internal class RetrofitCallAdapterFactory(
             throw IllegalArgumentException("Call return type must be parameterized as Call<Foo>")
         }
         val responseType: Type = getParameterUpperBound(0, returnType)
-        return RetrofitCallAdapter<Any>(responseType, chatParser)
+        return RetrofitCallAdapter<Any>(responseType, chatParser, callbackExecutor)
     }
 
     companion object {
-        fun create(chatParser: ChatParser) = RetrofitCallAdapterFactory(chatParser)
+        val mainThreadExecutor by lazy {
+            Executor { r ->
+                Handler(Looper.getMainLooper()).post(r)
+            }
```

Comment:
I'd bring this implementation closer to [what Retrofit has](https://github.com/square/retrofit/blob/7128bf34c649daca5977b1fbf93f0278fab4267c/retrofit/src/main/java/retrofit2/Platform.java#L138-L145), i.e. fetching the `Looper` and creating the `Handler` instance just once.

The `Handler` creation is relatively cheap (though it's still an object allocation), but grabbing the `Looper` is synchronized access of a static field.